File: projfabsoftcomercio/src/main/java/br/univille/projfabsoft/entity/Produto.java
package br.univille.projfabsoft.entity;

public class Produto {
    
    private Long id;
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }

    private String nome;
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }

    private String descricao;
    public String getDescricao() {
        return descricao;
    }
    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    private double preco;
    public double getPreco() {
        return preco;
    }
    public void setPreco(double preco) {
        this.preco = preco;
    }

    private int quantidadeEstoque;
    public int getQuantidadeEstoque() {
        return quantidadeEstoque;
    }
    public void setQuantidadeEstoque(int quantidadeEstoque) {
        this.quantidadeEstoque = quantidadeEstoque;
    }
}

File: projfabsoft_frontend/src/app/model/produto.ts
export class Produto {
  id?: number;
  nome: string = '';
  descricao: string = '';
  preco: number = 0;
  quantidadeEstoque: number = 0;
  imagemUrl?: string;

  constructor(
    id?: number,
    nome: string = '',
    descricao: string = '',
    preco: number = 0,
    quantidadeEstoque: number = 0,
    imagemUrl?: string
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.preco = preco;
    this.quantidadeEstoque = quantidadeEstoque;
    this.imagemUrl = imagemUrl;
  }
}
